### DISCUSSION TITLE
Porting mdk to Compose-multiplatform

### DISCUSSION THREAD

**Discussion by sbenmeddour** [wang-bin/mdk-sdk#244] [Show and tell]
Hello there

I'm working on a library like FVP, but for [Compose-multiplatform](https://www.jetbrains.com/lp/compose-multiplatform/), you can find it [here](https://github.com/sbenmeddour/mdk-compose)
I've almost done the iOS and Android part, there are probably a few bugs, or leaks, but this is still in early stage

Project is open-source, if you guys want to help or have any suggestion feel free to open an issue, or a PR

Actually I may need help on the desktop side (macOS, Windows): Compose runs on a JVM on desktop platforms, so to embed and use mdk-sdk native binaries, we'll have to create a JNI for it. If you have knowledge/experience with it feel free to explain me how to do it, or fork the project if you can

[I already made a JNI for Android](https://github.com/sbenmeddour/mdk-compose/tree/main/mdk-compose/src/androidMain/cpp/mdk-kotlin), but I am using Android's build system to compile it into a .so, i have no idea on how to do it for desktop (i'm not a c++ dev)

Thank you for reading me !

https://github.com/user-attachments/assets/ad71be0e-55b0-4733-8519-75ab603dc30d

https://github.com/user-attachments/assets/32f662ae-c39e-4eba-b371-082fa2412072

**Comment by DovgopolsSerj** [comment]
In fact, it's more correct to use the new [FFM API](https://docs.oracle.com/en/java/javase/23/core/foreign-function-and-memory-api.html) for the desktop. Because desktops are not limited to older versions of the JDK. And you won't need to compile native libraries.
It works faster and safer than JNI.

**Reply by sbenmeddour** [reply]
Seems interesting, will take a look.

I tried JNA before, but it required so much code I gave up, i switched to JNI which was a lot more simple to maintain because of compilation checks
